subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java

```diff
@@ -188,9 +180,7 @@ WatchingAwareVirtualFileSystem createVirtualFileSystem(
                 .orElse(new NonWatchingVirtualFileSystem(delegate));
             listenerManager.addListener(new VirtualFileSystemBuildLifecycleListener(
                 watchingAwareVirtualFileSystem,
-                StartParameter::isWatchFileSystem,
-                startParameter -> isSystemPropertyEnabled(VFS_DROP_PROPERTY, startParameter.getSystemPropertiesArgs()),
```

Comment:
Don't we want to keep the drop property?

Reply:
We do keep the drop property, we only don't pass it via a lambda to the `VirtualFileSystemBuildLifecycleListener`.

We now use a static method. We can go back to passing a lambda, though I don't remember why we did this in the first place.